examples/hosts/iframe-host/src/outer.ts

```diff
@@ -45,29 +37,23 @@ export async function loadFrame(iframeId: string, logId: string) {
         documentServiceFactory,
     });
 
-    const proxyContainer = await host.load(
-        { url: getDocumentUrl(documentId) },
-        iframe,
-    );
-
-    const text = document.getElementById(logId) as HTMLDivElement;
-    const quorum = proxyContainer.getQuorum();
-
-    const log =
-        (emitter: { on(event: string, listener: (...args: any[]) => void) }, name: string, ...events: string[]) => {
-            events.forEach((event) =>
-                // eslint-disable-next-line @typescript-eslint/no-unsafe-return
-                emitter.on(event, (...args) => {
-                    text.innerHTML += `${name}: ${event}: ${JSON.stringify(args)}<br/>`;
-                }));
-        };
-
-    quorum.getMembers().forEach((client) => text.innerHTML += `Quorum: client: ${JSON.stringify(client)}<br/>`);
-    log(quorum, "Quorum", "error", "addMember", "removeMember");
-    log(proxyContainer, "Container", "error", "connected", "disconnected");
+    await host.loadOuterProxy(iframe);
+
+    iframe.addEventListener("load", () => {
+        void (async () => {
+            // load the code inside the iframe but attach it here outside
+            const containerProxy: IContainerProxy =
+                await (iframe.contentWindow as any).loadContainer(documentId, createNew);
```

Comment:
>await (iframe.contentWindow as any).loadContainer(documentId, createNew); [](start = 16, length = 73)

this shouldn't work, at least we shouldn't need it to work as we should support cross origin and sandbox iframes where we have no code access to the internal frame, only post message

Reply:
Trying to make the postmessage proxy API bidirectional with comlink doesn't "just work", so I think this and restricting the iframe permissions might be better suited in a followup

---
In reply to: [530570016](https://github.com/microsoft/FluidFramework/pull/4404#discussion_r530570016) [](ancestors = 530570016)

Reply:
this only works because you are not sandboxing the iframe's code e.g. the code is running on the outer page. this defeats the purpose of iframeing

---
In reply to: [530660933](https://github.com/microsoft/FluidFramework/pull/4404#discussion_r530660933) [](ancestors = 530660933,530570016)